rosidl_generator_c/cmake/rosidl_generator_c_generate_interfaces.cmake

```diff
@@ -174,29 +174,46 @@ if(NOT rosidl_generate_interfaces_SKIP_INSTALL)
   ament_export_include_directories(include)
 endif()
 
-if(NOT "${_generated_msg_headers}${_generated_msg_sources}${_generated_srv_headers}${_generated_srv_sources} " STREQUAL " ")
-  find_package(ament_cmake_cppcheck)
-  if(ament_cmake_cppcheck_FOUND)
-    ament_cppcheck(
-      TESTNAME "cppcheck_rosidl_generated_c"
-      "${_output_path}")
-  endif()
-  find_package(ament_cmake_cpplint)
-  if(ament_cmake_cpplint_FOUND)
-    get_filename_component(_cpplint_root "${_output_path}" DIRECTORY)
-    ament_cpplint(
-      TESTNAME "cpplint_rosidl_generated_c"
-      # the generated code might contain longer lines for templated types
-      MAX_LINE_LENGTH 999
-      ROOT "${_cpplint_root}"
-      "${_output_path}")
-  endif()
-  find_package(ament_cmake_uncrustify)
-  if(ament_cmake_uncrustify_FOUND)
-    ament_uncrustify(
-      TESTNAME "uncrustify_rosidl_generated_c"
-      # the generated code might contain longer lines for templated types
-      MAX_LINE_LENGTH 999
-      "${_output_path}")
+if(BUILD_TESTING)
+  if(NOT "${_generated_msg_headers}${_generated_msg_sources}${_generated_srv_headers}${_generated_srv_sources} " STREQUAL " ")
+    find_package(ament_cmake_cppcheck QUIET)
+    if(ament_cmake_cppcheck_FOUND)
+      ament_cppcheck(
+        TESTNAME "cppcheck_rosidl_generated_c"
+        "${_output_path}")
+    else()
+      message(WARNING
```

Comment:
Something like this would be nice:

```
-- 'X' not found: skipping linting of generated message code
-- 'Y' not found: skipping linting of generated message code
-- One or more linters were not found. Please add them as a <test_depend> to enable linting.
```

The other option would be to just add `QUIET` to the `find_package`, but provide an option to `generate_messages` which switches that to `REQUIRED`. That way new message packages will not complain if the linters are not there, but we can ensure for a few of our core packages that the linters are found and run on the generated code.